### PR TITLE
Bug fix/drag and drop voting

### PR DESCRIPTION
**Pull request by jessemarek**
# Fixed Bugs with Dragon Drop Voting

----Description of the feature, what does it solve?----

* [x] Corrected wrong class name being applied to Drop Zone container when a drag was occurring.
* [x] Fixed a bug where the empty dragon would remain if a user started a drag and then dropped the dragon outside a valid drop zone.
* [x] Corrected `onDragEnd()` to properly handle logic if a destination was undefined.
* [x] Added type checking to `useRecoilState` hook for `DnDState` in `CastVoteContainer.tsx`.

### CODE REVIEW

File: src/state/DnDState/DnDAtoms.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { atom } from 'recoil';
 
-interface DnDContainerState {
+export interface DnDContainerState {
```
Comment (bramirez96) [inline src/state/DnDState/DnDAtoms.ts]: There are two variables exported on this module with the same name, that's A Bad Thing(tm)

File: src/components/pages/VotingPage/CastVote/CastVoteContainer.tsx [hunk 2]
```diff
@@ -5,7 +5,9 @@ import RenderCastVotes from './RenderCastVotes';
 import { DragDropContext, DragStart, DropResult } from 'react-beautiful-dnd';
 
 const CastVoteContainer = (): React.ReactElement => {
-  const [DnDState, setDnDState] = useRecoilState(DnD.DnDContainerState);
+  const [DnDState, setDnDState] = useRecoilState<DnD.DnDContainerState>(
```
Comment (bramirez96) [inline src/components/pages/VotingPage/CastVote/CastVoteContainer.tsx]: This shouldn't be necessary once you fixed the names exports issue

File: src/components/pages/VotingPage/DropZone/DropZone.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ const DropZone = ({
       {(provided: DroppableProvided, snapshot: DroppableStateSnapshot) => (
         <div
           className={`drop-zone${snapshot.isDraggingOver ? ' drag-over' : ''}${
-            snapshot.draggingFromThisWith ? ' drag-over' : ''
+            snapshot.draggingFromThisWith ? ' drag-from' : ''
```
Comment (bramirez96) [inline src/components/pages/VotingPage/DropZone/DropZone.tsx:22]: YES sorry I broke this lmao

**Review by bramirez96 — CHANGES_REQUESTED** [review]
Gotta fix a couple of things here

File: src/state/DnDState/DnDAtoms.ts (hunk 1 above)
Comment (jessemarek) [reply src/state/DnDState/DnDAtoms.ts]: Changed the names to be different

File: src/components/pages/VotingPage/CastVote/CastVoteContainer.tsx (hunk 2 above)
Comment (jessemarek) [reply src/components/pages/VotingPage/CastVote/CastVoteContainer.tsx]: I ended up removing the exporting of the interfaces altogether

File: src/components/pages/VotingPage/DropZone/DropZone.tsx (hunk 3 above)
Comment (jessemarek) [reply src/components/pages/VotingPage/DropZone/DropZone.tsx:22]: No worries we got it fixed and that's all that matters

**Review by bramirez96 — APPROVED** [review]

**Pull request merged**